src/serializer.ts

```diff
@@ -3,10 +3,18 @@ import { Transaction } from "../generated/transaction_pb";
 import { FiatAmount } from "../generated/fiat_amount_pb";
 import { XRPAmount } from "../generated/xrp_amount_pb";
 import { Currency } from "../generated/currency_pb";
+import Utils from "./utils";
 
 /* Allow `any` since this class doing progressive conversion of protocol buffers to JSON. */
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
+interface PaymentJSON {
+  Amount: object | string;
+  Destination: string;
+  DestinationTag?: number | undefined;
+  TransactionType: string;
```

Comment:
What `TransactionType`s are supported? If "Payment" is the only one, then I would do:
```
  TransactionType: "Payment";
```

Reply:
It's extensible, based on the `oneof` cases in the protocol buffers. At the moment, payment is the only supported option. 

I'm happy to hardcode this, but it makes the if statements more messy (because we still have to switch on a oneof) and the code is already written as is, so I'd prefer to leave it unless you have strong objections.

Reply:
It's fine to leave it. I don't understand this, though:

> we still have to switch on a oneof

Reply:
Sorry for not explaining. 

A oneof is a set of fields where only one will have a value: https://developers.google.com/protocol-buffers/docs/proto#oneof. The generated code guarantees this invariant. 

This is used in the following way:
- The `Transaction` proto has all the [Common Fields](https://xrpl.org/transaction-common-fields.html)
- The [`Transaction` proto](https://github.com/xpring-eng/xpring-common-protocol-buffers/blob/master/proto/transaction.proto) also has a `oneof` (`transaction_data`) field which has sub-messages for every type of Transaction supported. 
- Right now, we only support [`Payment`](https://github.com/xpring-eng/xpring-common-protocol-buffers/blob/master/proto/payment.proto), so the oneof is always set to Payment
- We should switch on which field is set: https://github.com/xpring-eng/xpring-common-js/blob/master/src/serializer.ts#L43

Happy to elaborate on any part of this.